mailserver/mailserver.go

```diff
@@ -189,6 +199,16 @@ func (s *WMailServer) DeliverMail(peer *whisper.Peer, request *whisper.Envelope)
 	}
 
 	if ok, lower, upper, bloom := s.validateRequest(peer.ID(), request); ok {
+		// Recover from possible goleveldb panics
+		defer func() {
```

Comment:
the deferred function will be defined in the scope of the function, and not of the `if` scope, so maybe it's better to move defer statement to the top of function to avoid possible confusion?

Reply:
The reason I put it there is that we only run that logic if we actually hit the DB, but I can move it to the top level scope.